src/components/SelectedItemBadge/SelectedItemBadge.tsx

```diff
@@ -158,82 +155,70 @@ export function SelectBadge<PriorityItemValue, OptionItem>({
                 </Dropdown>
             )}
 
-            {hasOptionList ? (
-                <Dropdown<OptionItem>
+            {children ? (
+                <Dropdown<ChildrenItemValue>
                     button={
                         <button
                             {...elem('optionButton', {
-                                isSelected: dropdownStates.option,
+                                isSelected: dropdownStates.main,
                             })}
-                            aria-label={option.buttonAriaLabel}
+                            {...(typeof label === 'string' && { 'aria-label': label })}
```

Comment:
Good catch. I added the optional `buttonLabel` prop back as it was before to better describe the button, since labels like `London` or `Manager` would not provide enough clarity about the button’s purpose. I also updated the story